ligo/src/types.mligo

```diff
@@ -0,0 +1,230 @@
+// Corresponds to Types.hs module
+
+// -- FA2 types -- //
+
+#if !TYPES_H
+#define TYPES_H
+
+type token_id = nat
+type token_metadata =
+  [@layout:comb]
+  { token_id : token_id
+  ; symbol : string
+  ; name : string
+  ; decimals : nat
+  ; extras : (string, string) map
+  }
+
+type operator =
+  { owner : address
+  ; operator : address
+  }
+type operators = (operator, unit) big_map
+
+type ledger_key = address * token_id
+type ledger_value = nat
+type ledger = (ledger_key, ledger_value) big_map
+
+// -- DAO base types -- //
+
+type migration_status =
+  [@layout:comb]
+  | Not_in_migration
+  | MigratingTo of address
+  | MigratedTo of address
+
+type proposal_key = bytes
+type proposal_metadata = (string, bytes) map
+type proposal =
+  [@layout:comb]
+  { upvotes : nat
+  ; downvotes : nat
+  ; start_date : timestamp
+  ; metadata : proposal_metadata
+  ; proposer : address
+  ; proposer_frozen_token : nat
+  ; voters : (address * nat) list
+  }
+
+type vote_type = bool
+
+type permit =
+  { key : key
+  ; signature : signature
+  }
+
+type contract_extra = (string, bytes) map
+
+// -- Storage -- //
+
+type storage =
+  { ledger : ledger
+  ; operators : operators
+  ; token_address : address
+  ; admin : address
+  ; pending_owner : address
+  ; migration_status : migration_status
+  ; voting_period : nat
+  ; quorum_threshold : nat
+  ; extra : contract_extra
+  ; proposals : (proposal_key, proposal) big_map
+  ; proposals_key_list_sort_by_date : proposal_key list
+  ; permits_counter : nat
+  }
+
+// -- Parameter -- //
+
+type transfer_destination =
+  [@layout:comb]
+  { to_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+type transfer_item =
+  [@layout:comb]
+  { from_ : address
+  ; txs : transfer_destination list
+  }
+type transfer_params = transfer_item list
+
+type balance_request_item =
+  [@layout:comb]
+  { owner : address
+  ; token_id : token_id
+  }
+type balance_response_item =
+  [@layout:comb]
+  { request : balance_request_item
+  ; balance : nat
+  }
+type balance_request_params =
+  [@layout:comb]
+  { requests : balance_request_item list
+  ; callback : balance_response_item list contract
+  }
+
+type token_metedata_registry_param = address contract
+
+type operator_param =
+  [@layout:comb]
+  { owner : address
+  ; operator : address
+  ; token_id : token_id
+  }
+type update_operator =
+  [@layout:comb]
+  | Add_operator of operator_param
+  | Remove_operator of operator_param
+type update_operators_param = update_operator list
+
+type fa2_parameter =
+    Transfer of transfer_params
+  | Balance_of of balance_request_params
+  | Token_metadata_registry of token_metedata_registry_param
+  | Update_operators of update_operators_param
+
+type transfer_ownership_param = address
```

Comment:
```suggestion
type transfer_ownership_param = [@annot:newOwner] address
```
Again only to match the Lorentz counterpart, I think this should work.

Reply:
Nope, it does not work :/

I suspect that LIGO cannot produce type annotations at all.